interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -711,7 +711,11 @@ namespace cling {
       }
     }
 
-    m_TransactionPool->releaseTransaction(&T);
+    if (!T.getScope()) {
+      // Do not unload a transaction for which an RAII is waiting to prevent
+      // double deletes (ROOT issue 7657).
+      m_TransactionPool->releaseTransaction(&T);
```

Comment:
Should this deletion prevention be here or inside `releaseTransaction`?  I.e. beside the actual delete is there any other action in `releaseTransaction` that should not be delayed until the end of the RAII?

Reply:
This comment of yours made me realize
- that this PR does not actually ever release transactions that were created by an RAII;
- that we can fix the issue at hand (and only that) by a way less intrusive change, #
- that we need to pass `unique_ptr<Transaction>` around to guarantee ownership and make it visible;

Reply:
> that this PR does not actually ever release transactions that were created by an RAII;

I think you might mean in "some" but not all code path (rolled-back might be one, other ends in "endTransaction" that will either push on poll or delete)